toucan_connectors/snowflake/snowflake_connector.py

```diff
@@ -119,10 +123,33 @@ def get_status(self) -> ConnectorStatus:
             with self.connect(login_timeout=5) as connection:
                 cursor = connection.cursor()
                 self._execute_query(cursor, 'SHOW WAREHOUSES', {})
-        except snowflake.connector.errors.ProgrammingError as e:
+
+                # Check if the default specified warehouse exists
+                res = self._execute_query(
+                    cursor, f"SHOW WAREHOUSES LIKE '{self.default_warehouse}'", {}
+                )
+                if res.empty:
+                    raise SnowflakeConnectorWarehouseDoesNotExists(
+                        f"The warehouse '{self.default_warehouse}' does not exists."
+                    )
+
+        except (
+            snowflake.connector.errors.OperationalError,
+            snowflake.connector.errors.ForbiddenError,
```

Comment:
comment tu as fait pour obtenir une forbidden ? 
je vois bien l'operationalerror lorsque je me trompe dans l'account.
Qu'est-ce qui pourrait provoquer une 403 en cas de mauvais `account` ?

Reply:
Si par exemple le user/mdp sont bons, mais que le user n'est pas autorisé à accéder à l'account spécifié

Reply:
J'ai déjà eu droit a des 403 sur l'auth (je crois me souvenir que c'est quand on se fait lock parce que trop de requêtes ont été faites dans un laps de temps très court)

Reply:
ok, donc je préfère quon ait un message différent pour une 403.
Là si qqun a une erreur 403, vérifier l'account ne va pas l'aider.

Plutot qque chose comme "Access forbidden, please check that you have access to this account or try later."

Reply:
Changé ici : https://github.com/ToucanToco/toucan-connectors/pull/376/commits/442183cd4151c6c675f02660bf5dca6cbdb3c37b